spec/tapioca/compilers/dsl/active_record_typed_store_spec.rb

```diff
@@ -0,0 +1,194 @@
+# typed: false
+# frozen_string_literal: true
+
+require "spec_helper"
+
+RSpec.describe(Tapioca::Compilers::Dsl::ActiveRecordTypedStore) do
+  describe("#initialize") do
+    def constants_from(content)
+      with_content(content) do
+        subject.processable_constants.map(&:to_s).sort
+      end
+    end
+
+    it("gathers no constants if there are no ActiveRecordTypedStore classes") do
+      expect(subject.processable_constants).to(be_empty)
+    end
+
+    it("gather only TypedStore classes") do
+      content = <<~RUBY
+        class Post < ActiveRecord::Base
+          typed_store :metadata do |s|
+            s.string(:reviewer, blank: false, accessor: false)
+          end
+        end
+        class Shop < ActiveRecord::Base
```

Comment:
What would happen if there was a `CustomPost < Post`?

Reply:
`["CustomePost", "Post"]` ... I will update the testcase.